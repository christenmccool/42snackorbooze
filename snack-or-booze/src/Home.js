import React from "react";
import { Card, CardBody, CardTitle, CardText } from "reactstrap";

function Home({snacks, drinks}) {
  return (
    <section className="col-md-8">
      <Card>
        <CardBody className="text-center">
          <CardTitle>
            <h3 className="font-weight-bold">
              Welcome to Silicon Valley's premier dive cafe!
            </h3>
          </CardTitle>
          <CardText className="font-italic">
                Serving {snacks.length} Different Snacks and {drinks.length} Different Drinks
          </CardText>
        </CardBody>
      </Card>
    </section>
  );
}

export default Home;
