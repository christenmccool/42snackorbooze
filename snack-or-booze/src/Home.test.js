import { render, asFragment, fireEvent, waitFor } from '@testing-library/react';
import Home from './Home';
import {drinks, snacks} from './mockTestDb';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';


test('renders without crashing', () => {
    render(<Home snacks={snacks} drinks={drinks}/>);
});

test("if matches snapshot", () => {
    const {asFragment} = render(<Home snacks={snacks} drinks={drinks}/>);
    expect(asFragment()).toMatchSnapshot();
})

test("if home renders", async () => {
    const { getByText, debug } = render(<Home snacks={snacks} drinks={drinks}/>);
    debug();

    expect(getByText(`Welcome to Silicon Valley's premier dive cafe!`)).toBeInTheDocument();
    expect(getByText(`Serving 3 Different Snacks and 4 Different Drinks`)).toBeInTheDocument();
})