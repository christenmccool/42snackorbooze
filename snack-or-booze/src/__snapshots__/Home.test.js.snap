// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-8"
  >
    <div
      class="card"
    >
      <div
        class="text-center card-body"
      >
        <h5
          class="card-title"
        />
        <h3
          class="font-weight-bold"
        >
          Welcome to Silicon Valley's premier dive cafe!
        </h3>
        <p
          class="font-italic card-text"
        >
          Serving 3 Different Snacks and 4 Different Drinks
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
